src/Compiler/Microsoft.AspNetCore.Razor.Language/test/TestFiles/ParserTests/CSharpBlockTest/EscapedIdentifiers_05.stree.txt

```diff
@@ -10,21 +10,22 @@
                     RazorMetaCode - [1..2)::1 - Gen<None>
                         LeftBrace;[{];
                     CSharpCodeBlock - [2..68)::66
-                        CSharpStatementLiteral - [2..68)::66 - [LF    var @string = "string test";LF    @@string = "new string";LF] - Gen<Stmt>
+                        CSharpStatementLiteral - [2..42)::40 - [LF    var @string = "string test";LF    ] - Gen<Stmt>
                             NewLine;[LF];
                             Whitespace;[    ];
                             Identifier;[var];
                             Whitespace;[ ];
-                            Transition;[@];
-                            Keyword;[string];
+                            Identifier;[@string];
                             Whitespace;[ ];
                             Assign;[=];
                             Whitespace;[ ];
                             StringLiteral;["string test"];
                             Semicolon;[;];
                             NewLine;[LF];
                             Whitespace;[    ];
+                        CSharpEphemeralTextLiteral - [42..43)::1 - [@] - Gen<None>
                             Transition;[@];
+                        CSharpStatementLiteral - [43..68)::25 - [@string = "new string";LF] - Gen<Stmt>
```

Comment:
So if you have two `@` symbols at the beginning of the line, the first `@` is ignored and then a normal C# statement is emitted? Is that an intended "feature"?

Reply:
They're not ignored. The first `@` escapes the second one, so you get `@string` in the final C# code. This is intended behavior.

Reply:
I saw CSharpEphemeralTextLiteral and thought that will be ignored, but what you say makes sense, thanks